apps/api/src/app/subscribers/usecases/search-by-external-subscriber-ids/search-by-external-subscriber-ids.use-case.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { IExternalSubscribersEntity, SubscriberEntity, SubscriberRepository } from '@novu/dal';
+import { SubscriberDto } from '@novu/shared';
+
+import { SearchByExternalSubscriberIdsCommand } from './search-by-external-subscriber-ids.command';
+
+@Injectable()
+export class SearchByExternalSubscriberIds {
+  constructor(private subscriberRepository: SubscriberRepository) {}
+
+  async execute(command: SearchByExternalSubscriberIdsCommand): Promise<SubscriberDto[]> {
+    const entity = this.mapToEntity(command);
+
+    const entities = await this.subscriberRepository.searchByExternalSubscriberIds(entity);
+
+    return entities.map(this.mapFromEntity);
+  }
+
+  private mapToEntity(command: SearchByExternalSubscriberIdsCommand): IExternalSubscribersEntity {
+    return {
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+      externalSubscriberIds: command.externalSubscriberIds,
```

Comment:
I'm wondering about two things here:
- why sometimes we do convert with `convertStringToObjectId` and sometimes not
- can we write a util that will do this conversion for us automatically, like `mapToEntity<Dto, Entity>(dto, ['environmentId', 'organizationId'])` like the second arg is an array of fields that should have names converted to `_name`

Reply:
About the first maybe a lack of consistency on my side due the huge amount of this work. I tried to apply this rule of thumb:
- for commands and dtos use a string therefore converting using `convertObjectIdToString`.
- for commands or queries to the DAL use already the ObjectId therefore converting using `convertStringToObjectId`

Regarding the second point I not fully totally understand but I already added a point to the Tech forum meeting regarding this functionalities that I added and if we wanted to expand them further, reshape them or use functionalities like the one you described. Basically I came with this quick implementation to handle transformations to DAL avoiding to creep a Mongoose import dependency into the API. That was my goal.